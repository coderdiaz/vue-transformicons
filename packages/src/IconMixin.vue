<script>
export default {
  props: {
    animation: {
      type: String,
      default: '',
    },
    active: {
      type: Boolean,
      default: false,
    },
  },
};
</script>
