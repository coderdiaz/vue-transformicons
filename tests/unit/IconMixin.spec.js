import { shallowMount } from '@vue/test-utils';
import IconMixin from '../../packages/src/IconMixin.vue';

const mockComponent = {
  template: '<div class="mock-component"></div>',
};

const vm = shallowMount(mockComponent, {
  mixins: [IconMixin],
});

describe('IconMixin', () => {
  it('should match the snapshot', () => {
    expect(vm.$el).toMatchSnapshot();
  });

  it('should set the mixin in the component', () => {
    const { animation, active } = vm.props();
    expect(animation).toBe('');
    expect(active).toBeFalsy();
  });
});
