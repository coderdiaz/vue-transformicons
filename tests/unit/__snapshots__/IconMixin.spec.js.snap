// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconMixin should match the snapshot 1`] = `undefined`;
